src/common/openssl_opts_handler.cc

```diff
@@ -128,15 +133,20 @@ void load_module(const string &engine_conf)
     log_error("failed to load modules from CONF:\n" + get_openssl_error());
   }
 }
+#endif // !OPENSSL_NO_ENGINE
 
 void init_engine()
 {
   string opts = g_ceph_context->_conf->openssl_engine_opts;
   if (opts.empty()) {
     return;
   }
+#ifdef OPENSSL_NO_ENGINE
+  derr << "OpenSSL is compiled with no engine, but openssl_engine_opts is set";
```

Comment:
I share @cbodley's worry. What was the behavior previously? Explicit failure at compile-time?